lib/xdr/ledger_entries/claimable_balance_entry.ex

```diff
@@ -0,0 +1,150 @@
+
+  @behaviour XDR.Declaration
+
+  @struct_spec XDR.Struct.new(
+                 claimable_balance_id: ClaimableBalanceID,
+                 claimant: Claimant,
+                 asset: Asset,
+                 amount: Int64,
+                 claimable_balance_entry_ext: ClaimableBalanceEntryExt
+               )
+
+  @type t :: %__MODULE__{
+          claimable_balance_id: ClaimableBalanceID.t(),
+          claimant: Claimant.t(),
+          asset: Asset.t(),
+          amount: Int64.t(),
+          claimable_balance_entry_ext: ClaimableBalanceEntryExt.t()
+        }
+
+  defstruct [:claimable_balance_id, :claimant, :asset, :amount, :claimable_balance_entry_ext]
+
+  @spec new(
+          claimable_balance_id :: ClaimableBalanceID.t(),
+          claimant :: Claimant.t(),
+          asset :: Asset.t(),
+          amount :: Int64.t(),
+          claimable_balance_entry_ext :: ClaimableBalanceEntryExt.t()
+        ) :: t()
+  def new(
+        %ClaimableBalanceID{} = claimable_balance_id,
+        %Claimant{} = claimant,
+        %Asset{} = asset,
+        %Int64{} = amount,
+        %ClaimableBalanceEntryExt{} = claimable_balance_entry_ext
+      ),
+      do: %__MODULE__{
+        claimable_balance_id: claimable_balance_id,
+        claimant: claimant,
+        asset: asset,
+        amount: amount,
+        claimable_balance_entry_ext: claimable_balance_entry_ext
+      }
+
+  @impl true
+  def encode_xdr(%__MODULE__{
+        claimable_balance_id: claimable_balance_id,
+        claimant: claimant,
+        asset: asset,
+        amount: amount,
+        claimable_balance_entry_ext: claimable_balance_entry_ext
+      }) do
+    [
+      claimable_balance_id: claimable_balance_id,
+      claimant: claimant,
+      asset: asset,
+      amount: amount,
+      claimable_balance_entry_ext: claimable_balance_entry_ext
+    ]
+    |> XDR.Struct.new()
+    |> XDR.Struct.encode_xdr()
+  end
+
+  @impl true
+  def encode_xdr!(%__MODULE__{
+        claimable_balance_id: claimable_balance_id,
+        claimant: claimant,
+        asset: asset,
+        amount: amount,
+        claimable_balance_entry_ext: claimable_balance_entry_ext
+      }) do
+    [
+      claimable_balance_id: claimable_balance_id,
+      claimant: claimant,
+      asset: asset,
+      amount: amount,
+      claimable_balance_entry_ext: claimable_balance_entry_ext
+    ]
+    |> XDR.Struct.new()
+    |> XDR.Struct.encode_xdr!()
+  end
+
+  @impl true
+  @spec decode_xdr(any, any) ::
+          {:error, :not_binary | :not_list}
+          | {:ok,
+             {%{
+                :__struct__ => StellarBase.XDR.ClaimableBalanceEntry | XDR.Struct,
+                optional(:amount) => map,
+                optional(:asset) => map,
+                optional(:claimable_balance_entry_ext) => map,
+                optional(:claimable_balance_id) => map,
+                optional(:claimant) => map,
+                optional(:components) => list
+              }, binary}}
```

Comment:
@einerzg It is not really clear why to use this spec instead of the XDR declaration. If the spec is really needed I suggest you keep it simple.

````
{:error, :not_binary | :not_list} | {:ok, t() | XDR.Struct.t()}

or

{:error, :atom()} | {:ok, struct()}